llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -3998,6 +3998,24 @@ void IndexCallsiteContextGraph::updateCall(CallInfo &CallerCall,
   CI->Clones[CallerCall.cloneNo()] = CalleeFunc.cloneNo();
 }
 
+// Update the debug information attached to NewFunc to use the clone Name. Note
+// this needs to be done for both any existing DISubprogram for the definition,
+// as well as any separate declaration DISubprogram.
+static void updateSubprogramLinkageName(Function *NewFunc, std::string Name) {
+  assert(Name == NewFunc->getName());
+  auto *SP = NewFunc->getSubprogram();
+  if (!SP)
+    return;
+  auto *NewName = MDString::get(NewFunc->getParent()->getContext(), Name);
```

Comment:
nit: Is this an MDString version of the Name? Maybe rename the var if so.

Reply:
done